.github/workflows/ci.yaml

```diff
@@ -0,0 +1,146 @@
+name: test
+on:
+  push:
+    branches:
+      - "main"
+      - "release-*"
+  pull_request:
+    branches: [ main ]
+jobs:
+  codegen:
+    name: Codegen
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Restore go build cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/go-build
+          key: ${{ runner.os }}-go-build-v1-${{ hashFiles('**/go.mod') }}
+      - name: Setup Golang
+        uses: actions/setup-go@v3.1.0
+        with:
+          go-version: "1.17"
+      - name: Add bins to PATH
+        run: |
+          echo /home/runner/go/bin >> $GITHUB_PATH
+          echo /usr/local/bin >> $GITHUB_PATH
+      - name: Install protoc
+        run: |
+          set -eux -o pipefail
+          PROTOC_VERSION=3.19.4
+          PROTOC_ZIP=protoc-$PROTOC_VERSION-linux-x86_64.zip
+          curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v$PROTOC_VERSION/$PROTOC_ZIP
+          sudo unzip -o $PROTOC_ZIP -d /usr/local bin/protoc
+          sudo unzip -o $PROTOC_ZIP -d /usr/local 'include/*'
+          sudo chmod +x /usr/local/bin/protoc
+          sudo find /usr/local/include -type f | xargs sudo chmod a+r
+          sudo find /usr/local/include -type d | xargs sudo chmod a+rx
+          rm -f $PROTOC_ZIP
+          ls /usr/local/include/google/protobuf/
+      - name: Install pandoc
+        run: |
+          set -eux -o pipefail
+          PANDOC_VERSION=2.17.1
+          PANDOC_ZIP=pandoc-$PANDOC_VERSION-linux-amd64.tar.gz
+          curl -OL https://github.com/jgm/pandoc/releases/download/$PANDOC_VERSION/$PANDOC_ZIP
+          sudo tar xvzf $PANDOC_ZIP --strip-components 1 -C /usr/local
+          rm -f $PANDOC_ZIP
+          echo /usr/local/pandoc-$PANDOC_VERSION/bin >> $GITHUB_PATH
+      - name: Get dependencies
+        run: go mod download
+      - name: Make codegen
+        run: |
+          echo 'GOPATH=/home/runner/go' >> $GITHUB_ENV
+          make -B codegen
+      - name: Ensure nothing changed
+        run: git diff --exit-code
+
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v3.1.0
+        with:
+          go-version: "1.17"
+        id: go
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+      - name: Restore Go build cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/go-build
+          key: ${{ runner.os }}-go-build-v1-${{ github.run_id }}
+      - name: Get dependencies
+        run: go mod download
+      - name: Test
+        run: make test
```

Comment:
`make test` won't run the test which has  labels`isb_jetstream` or `isb_redis`, should we call `make test-code`?

Reply:
udpated.